hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -617,9 +624,27 @@ long acquireLock() throws UnknownHostException, TException, InterruptedException
             Lists.newArrayList(lockComponent),
             System.getProperty("user.name"),
             InetAddress.getLocalHost().getHostName());
-    LockResponse lockResponse = metaClients.run(client -> client.lock(lockRequest));
-    AtomicReference<LockState> state = new AtomicReference<>(lockResponse.getState());
-    long lockId = lockResponse.getLockid();
+
+    // Only works in Hive 2 or later.
+    if (MetastoreVersion.min(MetastoreVersion.HIVE_2)) {
+      lockRequest.setAgentInfo(agentInfo);
+    }
+
+    Pair<Long, LockState> lockResult;
```

Comment:
Can we re-use the Tasks.retry().shouldRetryTest().run(...) like the rest of retries?

That way we get backoff and configurable number of retries.

Reply:
I also think we could have number of retries configurable, (and optionally backoff) otherwise it doesnt match how everything else is configurable.

Reply:
Let me check this out

Reply:
Thanks for the idea!
Done as suggested.